### PR TITLE
Added test in project. Coverage up to 100.

### PR DESCRIPTION
**Pull request by Mjh9122**

### CODE REVIEW

File: networkx/algorithms/bipartite/tests/test_project.py [hunk 1]
```diff
@@ -14,6 +14,10 @@ def test_path_projected_graph(self):
         P = bipartite.projected_graph(G, [0, 2])
         assert nodes_equal(list(P), [0, 2])
         assert edges_equal(list(P.edges()), [(0, 2)])
+        with pytest.raises(nx.NetworkXError):
+            G = nx.MultiGraph()
+            G.add_edge(0, 1)
+            bipartite.projected_graph(G, [0])
```
Comment (rossbar) [inline networkx/algorithms/bipartite/tests/test_project.py:20]: The best-practice here would be to only have the code that you expect to raise the warning within the context, the reason being that you can accidentally hit a NetworkXError from a branch you don't expect and would be none the wiser. Of course, using the `match=` kwarg helps with this as well, guaranteeing you're hitting the exact branch you think you're testing!

```suggestion
        G = nx.MultiGraph([(0, 1)])
        with pytest.raises(nx.NetworkXError, match="not defined"):
            bipartite.projected_graph(G, [0])
```

File: networkx/algorithms/bipartite/tests/test_project.py [hunk 2]
```diff
@@ -66,6 +69,14 @@ def test_path_weighted_projected_graph(self):
         assert edges_equal(list(P.edges()), [(0, 2)])
         P[0][2]["weight"] = 1
 
+    def test_digraph_conversion(self):
+        G = nx.DiGraph()
+        edges = [(0, 1, 1), (1, 2, 1), (2, 3, 1), (3, 4, 2)]
+        G.add_weighted_edges_from(edges)
```
Comment (rossbar) [inline networkx/algorithms/bipartite/tests/test_project.py:75]: Any reason to use weighted edges here? AFAICT edge weights in the original bipartite graph are not considered in the algorithm. The weights in the output graph are the result of the algorithm.

That actually might be a bit confusing as the node attributes are copied over from the input to the projected graph, but the edge attributes are not (and if there are weighted edges, the meaning of the "weight" attribute changes between the input and output graphs).

Comment (Mjh9122) [reply networkx/algorithms/bipartite/tests/test_project.py]: You're completely correct, made the change. Thanks.

**Review by MridulS — APPROVED** [review]

File: networkx/algorithms/bipartite/tests/test_project.py [hunk 3]
```diff
@@ -66,6 +69,12 @@ def test_path_weighted_projected_graph(self):
         assert edges_equal(list(P.edges()), [(0, 2)])
         P[0][2]["weight"] = 1
 
+    def test_digraph_conversion(self):
```
Comment (dschult) [inline networkx/algorithms/bipartite/tests/test_project.py:72]: Maybe a better test name would be: ???
```suggestion
    def test_digraph_weighted_projection(self):
```

**Review by rossbar — APPROVED** [review]
Thanks @Mjh9122 - the CI problems are unrelated so let's get this in!

**Pull request merged**